Build failure: droidcam-obs

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Build obs with `droidcam-obs` plugin.


### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/290045021

### Relevant log output

```console
nix log /nix/store/bm38nwcg3h7svs37naik5ly5ypgf3kry-droidcam-obs-2.0.2.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/bm38nwcg3h7svs37naik5ly5ypgf3kry-droidcam-obs-2.0.2.drv^*'
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/xk38hynywf0g4qj3sx6f4yzxmhdk2cig-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation droidcam-obs-2.0.2: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file './linux/linux.mk')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/fd118hwh7d1ncib4mdw56ylv3g9k0iyj-bash-5.2p37/bin/bash ALLOW_STATIC=no 
JPEG_DIR=/nix/store/r6l7p1j1fl4syzcxi7qchph2s7n0bgx6-libjpeg-turbo-3.0.4-dev JPEG_LIB=/nix/store/r8wi7x5qmr3ld2jj4bqk11gwf585z10a-libjpeg-turbo-3.0.4/lib 
IMOBILEDEV_DIR=/nix/store/xchjvcq36raffwi7697iz39zmqm7mbfy-libimobiledevice-1.3.0-unstable-2024-05-20
g++ -std=c++17 -Wall -fPIC -Isrc/ -I/nix/store/xchjvcq36raffwi7697iz39zmqm7mbfy-libimobiledevice-1.3.0-unstable-2024-05-20/include 
-I/nix/store/r6l7p1j1fl4syzcxi7qchph2s7n0bgx6-libjpeg-turbo-3.0.4-dev/include -I/nix/store/hdgkfddym117iib1w67dxayr54kp7b1s-ffmpeg-7.1-bin/include 
-I/nix/store/x3j22pk5yhh140s675bca798nb0qii20-obs-studio-31.0.1/include/obs -L/nix/store/r8wi7x5qmr3ld2jj4bqk11gwf585z10a-libjpeg-turbo-3.0.4/lib 
-L/nix/store/xchjvcq36raffwi7697iz39zmqm7mbfy-libimobiledevice-1.3.0-unstable-2024-05-20/lib -lturbojpeg -limobiledevice-1.0 -lobs -shared src/device_discovery.cc 
src/ffmpeg_decode.cc src/mdns_discovery.cc src/mjpeg_decode.cc src/net.cc src/plugin.cc src/proxy.cc src/sys/unix/cmd.cc  -o build/droidcam-obs.so
src/ffmpeg_decode.cc:21:10: fatal error: obs-ffmpeg-compat.h: No such file or directory
   21 | #include <obs-ffmpeg-compat.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
src/mdns_discovery.cc: In function 'int query_callback(int, const sockaddr*, size_t, mdns_entry_type_t, uint16_t, uint16_t, uint16_t, uint32_t, const void*, size_t, 
size_t, size_t, size_t, size_t, void*)':
src/mdns_discovery.cc:109:32: warning: 'in_addr' may be used uninitialized [-Wmaybe-uninitialized]
  109 |     fromaddrstr.str = inet_ntop(from->sa_family, in_addr, addrbuffer, (socklen_t)addrlen);
      |                       ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/mdns_discovery.cc:89:11: note: 'in_addr' was declared here
   89 |     void *in_addr;
      |           ^~~~~~~
make: *** [Makefile:42: build/droidcam-obs.so] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.2-Veil, NixOS, 25.05 (Warbler), 25.05.20250212.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, Nix alternative) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/xpciy4cpwxny54ijyag2zr4j3dsb915s-lix-2.91.1/share`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/wbwmnb9qcdbqqgav20pi72wabxnzpf19-source`


### Notify maintainers

@ulrikstrid 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
+++ b/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
@@ -10,22 +10,22 @@
   libplist,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "droidcam-obs";
   version = "2.0.2";
 
   src = fetchFromGitHub {
     owner = "dev47apps";
     repo = "droidcam-obs-plugin";
-    rev = version;
+    tag = finalAttrs.version;
     sha256 = "sha256-YtfWwgBhyQYx6QfrKld7p6qUf8BEV/kkQX4QcdHuaYU=";
   };
 
   postPatch = ''
     substituteInPlace ./linux/linux.mk \
-      --replace "-limobiledevice" "-limobiledevice-1.0" \
-      --replace "-I/usr/include/obs" "-I${obs-studio}/include/obs" \
-      --replace "-I/usr/include/ffmpeg" "-I${ffmpeg}/include"
+      --replace-fail "-limobiledevice" "-limobiledevice-1.0" \
+      --replace-fail "-I/usr/include/obs" "-I${obs-studio}/include/obs" \
+      --replace-fail "-I/usr/include/ffmpeg" "-I${ffmpeg}/include"
   '';
 
   preBuild = ''
@@ -61,12 +61,11 @@ stdenv.mkDerivation rec {
 
   doCheck = false;
 
-  meta = with lib; {
+  meta = {
     description = "DroidCam OBS";
     homepage = "https://github.com/dev47apps/droidcam-obs-plugin";
-    license = licenses.gpl2Plus;
-    maintainers = with maintainers; [ ulrikstrid ];
-    platforms = platforms.linux;
-    broken = true;
+    license = lib.licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ ulrikstrid ];
+    platforms = lib.platforms.linux;
   };
-}
+})

--- a/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
+++ b/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
@@ -8,8 +8,8 @@
   libimobiledevice,
   libusbmuxd,
   libplist,
+  pkg-config,
 }:
-
 stdenv.mkDerivation (finalAttrs: {
   pname = "droidcam-obs";
   version = "2.3.4";
@@ -34,13 +34,22 @@ stdenv.mkDerivation (finalAttrs: {
     ffmpeg
   ];
 
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  # Flag reference in regard to:
+  # https://github.com/dev47apps/droidcam-obs-plugin/blob/master/linux/linux.mk
   makeFlags = [
     "ALLOW_STATIC=no"
     "JPEG_DIR=${lib.getDev libjpeg}"
     "JPEG_LIB=${lib.getLib libjpeg}/lib"
-    "IMOBILEDEV_DIR=${libimobiledevice}"
+    "IMOBILEDEV_DIR=${lib.getDev libimobiledevice}"
+    "IMOBILEDEV_DIR=${lib.getLib libimobiledevice}"
     "LIBOBS_INCLUDES=${obs-studio}/include/obs"
     "FFMPEG_INCLUDES=${lib.getLib ffmpeg}"
+    "LIBUSBMUXD=libusbmuxd-2.0"
+    "LIBIMOBILEDEV=libimobiledevice-1.0"
   ];
 
   installPhase = ''

--- a/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
+++ b/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
@@ -12,21 +12,14 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "droidcam-obs";
-  version = "2.0.2";
+  version = "2.3.4";
 
   src = fetchFromGitHub {
     owner = "dev47apps";
     repo = "droidcam-obs-plugin";
     tag = finalAttrs.version;
-    sha256 = "sha256-YtfWwgBhyQYx6QfrKld7p6qUf8BEV/kkQX4QcdHuaYU=";
+    sha256 = "sha256-KWMLhddK561xA+EjvoG4tXRW4xoLil31JcTTfppblmA=";
   };
-
-  postPatch = ''
-    substituteInPlace ./linux/linux.mk \
-      --replace-fail "-limobiledevice" "-limobiledevice-1.0" \
-      --replace-fail "-I/usr/include/obs" "-I${obs-studio}/include/obs" \
-      --replace-fail "-I/usr/include/ffmpeg" "-I${ffmpeg}/include"
-  '';
 
   preBuild = ''
     mkdir ./build
@@ -46,6 +39,8 @@ stdenv.mkDerivation (finalAttrs: {
     "JPEG_DIR=${lib.getDev libjpeg}"
     "JPEG_LIB=${lib.getLib libjpeg}/lib"
     "IMOBILEDEV_DIR=${libimobiledevice}"
+    "LIBOBS_INCLUDES=${obs-studio}/include/obs"
+    "FFMPEG_INCLUDES=${lib.getLib ffmpeg}"
   ];
 
   installPhase = ''

--- a/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
+++ b/pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix
@@ -69,7 +69,7 @@ stdenv.mkDerivation (finalAttrs: {
     description = "DroidCam OBS";
     homepage = "https://github.com/dev47apps/droidcam-obs-plugin";
     license = lib.licenses.gpl2Plus;
-    maintainers = with lib.maintainers; [ ulrikstrid ];
+    maintainers = with lib.maintainers; [ NotAShelf ];
     platforms = lib.platforms.linux;
   };
 })

